packages/devtools_app/lib/src/screens/memory/panes/diff/diff_pane.dart

```diff
@@ -82,39 +26,37 @@ class DiffPane extends StatefulWidget {
 }
 
 class _DiffPaneState extends State<DiffPane> {
-  final controller = _DiffPaneController();
+  final controller = DiffPaneController();
 
   @override
   Widget build(BuildContext context) {
-    return DualValueListenableBuilder(
-      firstListenable: controller.snapshots,
-      secondListenable: controller.selectedIndex,
-      builder: (_, snapshots, index, __) {
+    return ValueListenableBuilder<int>(
+      valueListenable: controller.selectedIndex,
+      builder: (_, index, __) {
+        late Widget listContent;
+        final item = controller.selected;
+
+        if (item is InformationListItem) {
+          listContent = const _SnapshotDoc();
+        } else if (item is SnapshotListItem) {
+          listContent = _SnapshotContent(
+            item: item,
+            controller: controller,
+          );
+        } else {
+          throw Exception('Unexpected type of item: ${item.runtimeType}.');
+        }
+
         return Split(
           axis: Axis.horizontal,
           initialFractions: const [0.2, 0.8],
           minSizes: const [80, 80],
           children: [
             OutlineDecoration(
-              child: Column(
-                children: [
-                  _ListControlPane(controller: controller),
-                  Expanded(
-                    child: _SnapshotList(controller: controller),
-                  ),
-                ],
-              ),
+              child: SnapshotList(controller: controller),
```

Comment:
is it possible that the snapshots in the list will change and the selectedIndex stays the same? I think this only works right now because we change the index each time the list of snapshots changes, but if those were ever decoupled I think this would break. 

When we discussed this earlier, I was saying that the list content only needs to rebuild when the selected index changes - not when the list of snapshots changes. But the SnapshotList itself probably needs to be notified of both. You could put a ValueListenableBuilder inside SnapshotList.build that listens for the list of snapshots.